File: web/src/api/film.js
import request from '@/utils/request'

export function getFilmList(params) {
  return request({
    url: '/film/list',
    method: 'get',
    params
  })
}

// 新增电影
export function createMovie(data) {
  return request({
    url: '/film/create',
    method: 'post',
    data
  })
}

// 删除电影
export function deleteMovie(id) {
  return request({
    url: `/film/delete/${id}`,
    method: 'delete'
  })
}

export function getActors() {
  return request({
    url: '/film/actors',
    method: 'get'
  })
}

// 电影详情获取方法
export function getMovieDetail(id) {
  return request({
    url: `/film/detail/${id}`,
    method: 'get'
  })
}

// 电影更新方法
export function updateMovie(data) {
  const id = data.id;
  return request({
    url: `/film/edit/${id}`,
    method: 'put',
    data
  })
}

File: server/router/film/film.go
package film

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

type FilmRouter struct{}

func (s *FilmRouter) InitFilmRouter(Router *gin.RouterGroup) {
	filmRouter := Router.Group("film")
	var filmApi = v1.ApiGroupApp.FilmApiGroup.FilmApi
	{
		filmRouter.GET("list", filmApi.GetFilmList)          // 电影列表接口
		filmRouter.POST("create", filmApi.CreateMovie)       // 创建电影
		filmRouter.GET("actors", filmApi.GetActors)          // 新增演员列表路由
		filmRouter.DELETE("delete/:id", filmApi.DeleteMovie) // 删除接口
		filmRouter.GET("detail/:id", filmApi.GetMovieDetail) // 电影详情接口
		filmRouter.PUT("edit/:id", filmApi.UpdateMovie)      // 电影更新接口
	}
}
